### PR TITLE
DPL: avoid warning about uninitialised usage

### PR DESCRIPTION
**Pull request by ktf**
Still not clear to me wether the condition can actually be triggered.

It might explain phantom channels to the input proxy.

### CODE REVIEW

**Comment by ktf** [comment]
@TimoWilken macOS fails without logs.

**Comment by ktf** [comment]
@TimoWilken I still regularly see PRs being sent back in the queue.

File: Framework/Core/src/DeviceSpecHelpers.cxx [hunk 1]
```diff
@@ -708,14 +722,16 @@ void DeviceSpecHelpers::processInEdgeActions(std::vector<DeviceSpec>& devices,
   for (size_t edge : inEdgeIndex) {
     auto& action = actions[edge];
 
-    size_t consumerDevice;
+    size_t consumerDevice = -1;
```
Comment (anerokhi) [inline Framework/Core/src/DeviceSpecHelpers.cxx:725]: Is this necessary with `else` branch added?

Comment (ktf) [reply Framework/Core/src/DeviceSpecHelpers.cxx:725]: Not necessary, but allows us to spot actual issues if the code is changed.

**Pull request merged**